truss/templates/server/common/truss_server.py

```diff
@@ -145,20 +141,39 @@ async def predict(
             if self.is_binary(request):
                 with tracing.section_as_event(span, "binary-deserialize"):
                     body = truss_msgpack_deserialize(body_raw)
+                if model.truss_schema:
+                    try:
+                        with tracing.section_as_event(span, "parse-pydantic"):
+                            body = model.truss_schema.input_type.parse(**body)
```

Comment:
non-blocking, and I also don't think we have very many examples of this anyway, but would be good to have a test in test_model_inference for this case (binary w pydantic typing)

Reply:
Good point, that actually helped find a bug!